### PR TITLE
Add Keebfront Vanguard65 Stripped Down Firmware

### PR DESCRIPTION
**Pull request by mrnoisytiger**
## Description

Add Keebfront Vanguard65 stripped down firmware. Closing https://github.com/qmk/qmk_firmware/pull/19762 in connection with this pull request. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebfront/vanguard65/info.json [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+    "manufacturer": "Keebfront",
+    "keyboard_name": "Vanguard65",
+    "maintainer": "Keebfront",
+    "bootloader": "atmel-dfu",
+    "build": {
+        "lto": true
+    },
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "enabled": true,
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/info.json:11]:
```suggestion
```

File: keyboards/keebfront/vanguard65/info.json [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+        "rotary": [
+            {"pin_a": "E6", "pin_b": "D4"}
+        ]
+    },
+    "features": {
+        "extrakey": true,
+        "midi": true,
+        "rgblight": true
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/info.json:19]:
```suggestion
        "rgblight": true,
        "encoder": true
```

File: keyboards/keebfront/vanguard65/config.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+/* Define MIDI */
+#define MIDI_ADVANCED
+#define SLIDER_PINA D6
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/config.h:22]:
```suggestion
#define MIDI_ADVANCED
```

File: keyboards/keebfront/vanguard65/keymaps/via/rules.mk [hunk 4]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
+SRC += analog.c
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/keymaps/via/rules.mk:3]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
In the future, please avoid closing and opening new pull requests.

**Review by drashna — APPROVED** [review]

**Comment by mrnoisytiger** [comment]
Bump 😊

File: keyboards/keebfront/vanguard65/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+#include QMK_KEYBOARD_H
+
+enum layers {
+    _LAYER0,
+    _LAYER1
+};
+
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/keymaps/via/keymap.c:24]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/keebfront/vanguard65/info.json [hunk 6]
```diff
@@ -0,0 +1,743 @@
...
+    },
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "rotary": [
+            {"pin_a": "E6", "pin_b": "D4"}
+        ]
+    },
+    "features": {
+        "extrakey": true,
+        "midi": true,
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/info.json:17]: But no midi code/keycodes used?

File: keyboards/keebfront/vanguard65/config.h
Comment (fauxpark) [inline keyboards/keebfront/vanguard65/config.h:1]: This file can be deleted.

File: keyboards/keebfront/vanguard65/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+    [_LAYER1] = LAYOUT_all(
+        _______, QK_GESC, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, KC_END, 
+                 _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SCRL, KC_PAUS, QK_BOOT, _______,
+                 _______, RGB_SPI, RGB_SPD, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
+                 _______, _______, _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU,  
+                 _______, _______, _______,                            _______,                   _______, _______, KC_MPRV, KC_VOLD, KC_MNXT
+    ),
+
+};
```
Comment (fauxpark) [inline keyboards/keebfront/vanguard65/keymaps/via/keymap.c:48]:
```suggestion
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {

    [_LAYER0] = LAYOUT_all(
        KC_MUTE, QK_GESC, KC_1,    KC_2,    KC_3, KC_4, KC_5, KC_6,  KC_7, KC_8, KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL, KC_HOME,
                 KC_TAB,  KC_Q,    KC_W,    KC_E, KC_R, KC_T, KC_Y,  KC_U, KC_I, KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,         KC_PGUP, 
                 KC_CAPS, KC_A,    KC_S,    KC_D, KC_F, KC_G, KC_H,  KC_J, KC_K, KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                   KC_PGDN, 
                 KC_LSFT, KC_NUBS, KC_Z,    KC_X, KC_C, KC_V, KC_B,  KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP, 
                 KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,            KC_RALT, MO(1),   KC_LEFT, KC_DOWN, KC_RGHT
    ),

    [_LAYER1] = LAYOUT_all(
        _______, QK_GESC, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, KC_END, 
                 _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SCRL, KC_PAUS, QK_BOOT, _______,
                 _______, RGB_SPI, RGB_SPD, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
                 _______, _______, _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU,  
                 _______, _______, _______,                            _______,                   _______, _______, KC_MPRV, KC_VOLD, KC_MNXT
    ),

};

#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [_LAYER0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
    [_LAYER1] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)}
};
#endif
```
Just by convention.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**